### PR TITLE
Use new remoteStorage no-global build and widget

### PR DESCRIPTION
**Pull request by galfert**

### CODE REVIEW

**Comment by skddc** [comment]
Yay!

**Comment by skddc** [comment]
Yay!

Just one question: does this include a fix for the widget problem you experienced? If not, we can't deploy it to production, right? (In which case I'd prefer not to merge it to master until we can.)

**Comment by galfert** [comment]
RIght, I forgot to mention that I was not able to reproduce the problem ever again after I first encountered it. So I was not able to fix it yet.

But maybe we can deploy this version to a separate URL, so we can at least use it ourselves. Perhaps we then find out in which situations the problem occurs.

**Comment by skddc** [comment]
Yes, I'll deploy it to https://webmarks-dev.5apps.com/ tomorrow.

**Comment by galfert** [comment]
Did you deploy this yet?

**Comment by skddc** [comment]
No. I can't deploy it with the current script, as there's only one build branch, so it assumes that everything deployed on `-dev` is being tested and fixed before production can be deployed again. I have to change it and add another app (or you could just create one and deploy a subtree from some branch without pushing to this repo).

**Comment by galfert** [comment]
I deployed it to https://webmarks-ng.5apps.com. The "ng" stands for "no-global" or "next generation", what ever you fancy more :)

Will change my bookmarklet accordingly and use it regularly to see if any problems occur.

**Comment by skddc** [comment]
Cool!

It's also a bit of an ironic name for an Ember app, as `-ng` is usually used to mark Angular things. ;)

File: app/styles/app.scss [hunk 1]
```diff
@@ -305,10 +305,10 @@ section#welcome {
 
 // REMOTESTORAGE WIDGET
 
-#remotestorage-connect {
-  top: 0;
-  right: 0;
+#remotestorage-widget {
```
Comment (skddc) [inline app/styles/app.scss]: Why this change, if it's attached to `#remotestorage-connect` above?

File: app/services/storage.js [hunk 2]
```diff
@@ -94,14 +111,20 @@ export default Ember.Service.extend(Ember.Evented, {
   },
 
   setupRemoteStorage() {
+    const remoteStorage = this.get('remoteStorage');
```
Comment (skddc) [inline app/services/storage.js]: Is there a reason to leave the remoteStorage instance creation in `init()`, if it's not being configured entirely there?

Comment (galfert) [inline app/services/storage.js]: No. But actually the `setup()` call to configure it should be there as well. Then we can get rid of the whole `app/instance-initializers/storage.js`.

Comment (skddc) [inline app/services/storage.js]: I think there's a reason for that, because when you move it, the initial render jumps from disconnected to connected.

**Comment by skddc** [comment]
I pushed my latest changes (using the new module loading API, removing the instance initializer, adding a fetch method, so that `loadBookmarks` really just loads them into the collection and we can implement timeouts for bad connections. I tried adding a timeout already, but it didn't work for some reason. Have to check some other code where we're doing that.

**Comment by skddc** [comment]
The tests are failing because the xmlhttprequest module was removed from rs.js deps:

> Error: Cannot find module 'xmlhttprequest' from '/home/ubuntu/webmarks/node_modules/remotestoragejs/release/stable'

**Comment by skddc** [comment]
Rebased and merged locally. Thanks! ❤️
